.github/.scala-steward.conf

```
@@ -0,0 +1,5 @@
+updates.pin = [
+  { groupId = "org.scala-lang", artifactId="scala3-library", version = "3.1." },
+  { groupId = "org.scala-lang", artifactId="scala3-library_sjs1", version = "3.1." },
+  { groupId = "org.scala-js", artifactId="sbt-scalajs", version = "1.10." },
```

Shoot, these commas make trouble don't they.
```suggestion
  { groupId = "org.scala-js", artifactId="sbt-scalajs", version = "1.10." }
```

---

We need an action to validate these.

---

I have a linter in Emacs built on pyhocon. I don't think a Python dependency is the right public solution, but it's a frequent enough pain I support an alternate solution.

---

I have a linter in Emacs built on pyhocon. I don't think a Python dependency is the right public solution, but it's a frequent enough pain I support an alternate solution.

---

I mean, ideally steward itself would parse it so it's more than valid syntax, but actually valid config.

---

Yep, agreed.  I've never seen a HOCON schema definition.  I think it would have to be a small CLI based on the Steward code itself.

---

A scala-cli script like this could do it:
```scala
//> using scala "2"
import $dep.`org.scala-steward::scala-steward-core:0.14.0`

import org.scalasteward.core.repoconfig.RepoConfigAlg

val content = scala.io.Source.fromFile(args(0)).mkString
RepoConfigAlg.parseRepoConfig(content) match {
  case Left(err) => throw new Throwable(err)
  case Right(_) =>
}
```

> I mean, ideally steward itself would parse it so it's more than valid syntax, but actually valid config.

Do mean integrating the above into Scala Steward itself and then calling it with arguments like `--validate-repo-config $config.file` would only do the validation of `$config.file` and nothing else? Sounds good to me.

---

Nice! That would be fantastic. Then we can package it up into an action too.